### PR TITLE
fix: code optimalizations and doc update

### PR DESCRIPTION
**Pull request by Glenn-Terjesen**

### CODE REVIEW

File: workflow-templates/gradle-build-sonar-scan.yml [hunk 1]
```diff
@@ -88,13 +98,16 @@ jobs:
...
         run: |
             ./gradlew jacocoTestReport sonar --no-scan --no-rebuild -x test \
-                -Dsonar.projectKey=${{ env.SONAR_PROJECT_KEY }} \
+                -Dsonar.projectKey=${SONAR_PROJECT_KEY} \
                 -Dsonar.organization=enturas-github \
                 -Dsonar.host.url=https://sonarcloud.io \
-                -Dsonar.branch.name=${{ github.ref_name }} \
-                -Dsonar.token=${{ secrets.ENTUR_SONAR_PASSWORD }}
+                -Dsonar.branch.name=${BRANCH_NAME} \
+                -Dsonar.token=${ENTUR_SONAR_PASSWORD}
```
Comment (Chr1stian) [inline workflow-templates/gradle-build-sonar-scan.yml:110]: Hvor/hvordan settes denne nå?

Comment (Glenn-Terjesen) [reply workflow-templates/gradle-build-sonar-scan.yml:110]: for å hindre "injection attacks" går vi omveien via environment variabler,  det settes rett over :-)

File: workflow-templates/gradle-build-sonar-scan.yml [hunk 2]
```diff
@@ -88,13 +98,16 @@ jobs:
         env:
           ARTIFACTORY_USER: ${{ secrets.ARTIFACTORY_USER }}
           ARTIFACTORY_APIKEY: ${{ secrets.ARTIFACTORY_APIKEY }}
+          SONAR_PROJECT_KEY: ${{ env.SONAR_PROJECT_KEY }}
+          BRANCH_NAME: ${{ github.ref_name }}
+          ENTUR_SONAR_PASSWORD: ${{ secrets.ENTUR_SONAR_PASSWORD }}
```
Comment (Glenn-Terjesen) [inline workflow-templates/gradle-build-sonar-scan.yml:103]: @Chr1stian her :-)

Comment (Chr1stian) [reply workflow-templates/gradle-build-sonar-scan.yml:103]: Så ikke denne 😅

**Pull request merged**